### PR TITLE
Clang-tidy fixes for NUClear Roles

### PR DESCRIPTION
**Pull request by TrentHouliston**
Applies fixes to the NUClear roles code so that it conforms to some of the clang-tidy rules. Also removes some files that should have been deleted as they werenever used/not used anymore

### CODE REVIEW

File: nuclear/message/generator/Field.py [hunk 1]
```diff
@@ -92,6 +92,9 @@ def get_cpp_type_info(self):
         vector_regex = re.compile(r"^\.([fiuc]?)vec(\d*)$")
         matrix_regex = re.compile(r"^\.([fiuc]?)mat(\d*)$")
 
+        # Nothing is trivially copyable unless it is
```
[the remark this answers is not in the document]
Comment (TrentHouliston) [reply nuclear/message/generator/Field.py:95]: Best comment ever!

File: nuclear/message/generator/Message.py [hunk 2]
```diff
@@ -44,16 +44,25 @@ def generate_default_constructor(self):
...
-                    for v in self.fields
-                ]
-            )
-            field_set = ", ".join(["{0}(_{0})".format(v.name) for v in self.fields])
+            # Basic types are const reference and others are copy/moved
+            field_list = []
+            default_field_list = []
+            field_set = []
+            for v in self.fields:
+                # Work out if we should be copying a const reference or moving a trivial type
```
[the remark this answers is not in the document]
Comment (TrentHouliston) [reply nuclear/message/generator/Message.py:52]: You're right, this comment is screwed up. We move the non trivial types and copy the const references to trivial ones
```suggestion
                # Work out if we should be copying a trivial type from a const reference or moving a more complicated type
```

**Comment by TrentHouliston** [comment]
When this is ready i'll run ./b format on it to fix up all the formatting problems that this created. But until then that'll just muddy the diff

**Review by KipHamiltons — APPROVED** [review]
LGTM

**Review by KipHamiltons — APPROVED** [review]

**Pull request merged**